TargetComposition/SymBuilder/Readme.txt

```diff
@@ -563,6 +674,188 @@ Set bu breakpoint
 
 (00007ffd`29db0100)   ntdll!test+0x9   
 
+//*************************************************
+// STARTING OUT: A SECOND MORE COMPLEX EXAMPLE
+//*************************************************
+
+Consider another example of debugging notepad after the introduction of symbol builder symbols:
+
+0:000> .load SymbolBuilderComposition.dll
+0:000> dx @$s = Debugger.Utility.SymbolBuilder.CreateSymbols("notepad.exe")
+@$s = Debugger.Utility.SymbolBuilder.CreateSymbols("notepad.exe")                
+    Data            
+    Functions       
+    Types           
+0:000> .reload
+........................................
+0:000> k3
+ # Child-SP          RetAddr               Call Site
+00 000000b5`3dd6fab8 00007ff9`27b51b3e     win32u!ZwUserGetMessage+0x14
+01 000000b5`3dd6fac0 00007ff7`1b7ab020     user32!GetMessageW+0x2e
+02 000000b5`3dd6fb20 00007ff7`1b7c3ec6     notepad+0xb020
+
+//
+// Let's introduce a function for notepad!wWinMain:
+//
+0:000> dx @$wWinMain = @$s.Functions.Create("wWinMain", "int", 0xad6c, 0x3bc)
+@$wWinMain = @$s.Functions.Create("wWinMain", "int", 0xad6c, 0x3bc)                 : Function: wWinMain

```

Comment:
>0x3bc [](http://example.com/codeflow?start=61&length=5)

Where did the code size come from? Manual analysis? I think it would be fun to create something on top of this that consumed data from the disassembler or pdata. (I should add pdata processing to ImageInfo...)

Reply:
That came from me looking at the private symbols.  It can also be discovered by simply looking at the disassembly by hand.  

That said, I have another branch I've been working on forked on this PR which I will push to PR *AFTER* this merges which does, in fact, have a method

.PromoteToFunction()

which will use the data model disassembler to find the extents of the function by doing the standard .DisassembleFunction and looking at the address ranges of all basic blocks ;)

Reply:
SO COOL. Can't wait!

Reply:
Although I still kind of want to do the pdata thing for binaries without public symbols...